packages/vscode/src/plugins/prisma-language-server/index.ts

```diff
@@ -73,21 +89,39 @@ const startFileWatcher = (rootPath: string) => {
     depth: 9,
     recursive: true,
     ignoreInitial: true,
-    ignore: (targetPath) => {
-      if (targetPath === rootPath) {
-        return false
+    ignore(targetPath) {
+      if (!targetPath.startsWith(rootPath)) return true
+      if (targetPath.endsWith('.git')) return true
+
+      // every time we hit a project path, ensure some sub-paths are whitelisted
+      if (!(targetPath in allowedWatcherPaths) && fs.statSync(targetPath).isDirectory()) {
+        try {
+          // we get the location of installation of the @prisma/client package
+          const clientPath = path.dirname(require.resolve('@prisma/client', { paths: [targetPath] }))
+          const dotClientPath = path.resolve(clientPath, '..', '..', '.prisma', 'client', 'index.d.ts')
+
+          // if we have seen this path already, we don't need to save it again
+          if (allowedWatcherPaths[dotClientPath] !== undefined) return false
+          const dotClientArrayPath = dotClientPath.split(path.sep)
+
+          // decompose the path into an all the possible steps of its sub-paths
+          // eg. "a/b/c/d"" will become [ "a", "a/b", "a/b/c", "a/b/c/d" ]
```

Comment:
Required by the library, otherwise it either watches everything or ignores everything. This is the only way to get it granular :/